File: app/src/main/java/com/jeffrey/plashdom/service/InternetConnectionService.java
package com.jeffrey.plashdom.service;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public class InternetConnectionService {

    public static boolean checkConnection(Context context) {
        final ConnectivityManager connMgr = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);

        if (connMgr != null) {
            NetworkInfo activeNetworkInfo = connMgr.getActiveNetworkInfo();

            if (activeNetworkInfo != null) { // подключено к интернету
                // подключение через симку
                if (activeNetworkInfo.getType() == ConnectivityManager.TYPE_WIFI) {
                    // подключение через wifi
                    return true;
                }
                else return activeNetworkInfo.getType() == ConnectivityManager.TYPE_MOBILE;
            }
        }
        return false;
    }
}

File: app/src/main/java/com/jeffrey/plashdom/service/InternetConnectionService.kt
package com.jeffrey.plashdom.service

import android.content.Context
import android.net.ConnectivityManager

object InternetConnectionService {
    fun checkConnection(context: Context): Boolean {
        val connMgr = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val activeNetworkInfo = connMgr.activeNetworkInfo
        if (activeNetworkInfo != null) {
            return if (activeNetworkInfo.type == ConnectivityManager.TYPE_WIFI) {
                true
            } else activeNetworkInfo.type == ConnectivityManager.TYPE_MOBILE
        }
        return false
    }
}